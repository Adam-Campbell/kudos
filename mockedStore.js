export const store = {
    articles: {
        article_A: {
            _id: 'article_A',
            title: 'The title for Article A',
            description: 'The description for Article A.',
            image: 'https://example.com/article_a.jpg',
            text: 'The main body of Article A.',
            kudos: 7,
            author: 'user_A',
            category: 'category_A',
            commentIds: ['comment_A', 'comment_C']
        },
        article_B: {
            _id: 'article_B',
            title: 'The title for Article B',
            description: 'The description for Article B.',
            image: 'https://example.com/article_b.jpg',
            text: 'The main body of Article B.',
            kudos: 2,
            author: 'user_B',
            category: 'category_B',
            commentIds: ['comment_B', 'comment_D']
        },
        article_C: {
            _id: 'article_C',
            title: 'The title for Article C',
            description: 'The description for Article C.',
            image: 'https://example.com/article_c.jpg',
            text: 'The main body of Article C.',
            kudos: 0,
            author: 'user_A',
            category: 'category_B',
            commentIds: []
        }
    },
    users: {
        user_A: {
            _id: 'user_A',
            followers: 1,
            following: 0,
            username: 'Joe Bloggs',
            avatar: 'https://example.com/user_a.jpg',
            bio: 'The bio for User A.',
            postIds: ['article_A', 'article_C'],
            commentIds: ['comment_A', 'comment_B'],
            kudosIds: ['article_A'],
            highlightIds: ['article_A']
        },
        user_B: {
            _id: 'user_B',
            followers: 0,
            following: 1,
            username: 'Jane Smith',
            avatar: 'https://example.com/user_b.jpg',
            bio: 'The bio for User B.',
            postIds: ['article_B'],
            commentIds: ['comment_C'],
            kudosIds: ['article_B']
        }
    },
    comments: {
        comment_A: {
            _id: 'comment_A',
            text: 'The text for Comment A.',
            author: 'user_A',
            discussion: 'article_A',
            createdAt: 'Wed, 06 Jun 2018 08:07:53 GMT',
            parents: ['comment_A']
        },
        comment_B: {
            _id: 'comment_B',
            text: 'The text for Comment B.',
            author: 'user_A',
            discussion: 'article_B',
            createdAt: 'Wed, 06 Jun 2018 10:32:06 GMT',
            parents: ['comment_B']
        },
        comment_C: {
            _id: 'comment_C',
            text: 'The text for Comment C.',
            author: 'user_B',
            discussion: 'article_A',
            createdAt: 'Wed, 06 Jun 2018 12:21:27 GMT',
            parents: ['comment_A', 'comment_B']
        },
        comment_D: {
            _id: 'comment_D',
            text: 'Comment deleted by user.',
            discussion: 'article_B',
            createdAt: 'Wed, 06 Jun 2018 13:46:12 GMT',
            author: null,
            discussion: 'article_B',
            parents: ['comment_D']
        }
    },
    highlights: {
        highlight_A: {
            _id: 'highlight_A',
            post: 'article_A',
            user: 'user_A',
            excerpt: 'An excerpt from Article A.'
        }
    },
    currentUser: {
        isLoggedIn: true,
        token: 'abc123',
        _id: 'user_A',
        follows: ['user_B'],
        email: 'example@example.com'
    },
    categories: {
        all: ['article_A', 'article_B', 'article_C'],
        category_A: ['article_A'],
        category_B: ['article_B', 'article_C']
    },
    errors: {
        signUpDuplicateError: false,
        emailNotFound: false
    },
    successes: {
        passwordResetEmailSent: false
    }
}