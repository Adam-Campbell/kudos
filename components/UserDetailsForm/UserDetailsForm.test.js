import { shallow } from 'enzyme';
import { store } from '../../mockedStore';
import { UserDetailsFormContainer } from './index';
const mockedUpdateUserDetails = jest.fn();

describe('<UserDetailsFormContainer/>', () => {
    test('renders correctly', () => {
        const container = shallow(
            <UserDetailsFormContainer 
                users={store.users}
                currentUser_id={store.currentUser._id}
                token={store.currentUser.token}
                email={store.currentUser.email}
                updateUserDetails={mockedUpdateUserDetails}
            />
        );
        // create wrapper for presentational component returned
        const presenter = container.first().shallow();
        // check snapshots for container and presentational components
        expect(container).toMatchSnapshot();
        expect(presenter).toMatchSnapshot();
    });
    test('controlled inputs interact with state as expected', () => {
        const container = shallow(
            <UserDetailsFormContainer 
                users={store.users}
                currentUser_id={store.currentUser._id}
                token={store.currentUser.token}
                email={store.currentUser.email}
                updateUserDetails={mockedUpdateUserDetails}
            />
        );
        const presenter = container.first().shallow();
        // test that container state and controlled input value are both set to the initial state
        expect(container.state('username')).toBe('Joe Bloggs');
        expect(presenter.find('Input#username').props().value).toBe('Joe Bloggs');
        // update the controlled input
        presenter.find('Input#username').simulate('change', { target: { value: 'John Smith' } });
        // test that the state updates when the controlled input does.
        expect(container.state('username')).toBe('John Smith');
        // force rerender of container and create wrapper for the new presentational component
        container.update();
        const updatedPresenter = container.first().shallow();
        // test that the new presentational component is rendered with the new state
        expect(updatedPresenter.find('Input#username').props().value).toBe('John Smith');
    });
    test('the form submits as expected', () => {
        const container = shallow(
            <UserDetailsFormContainer 
                users={store.users}
                currentUser_id={store.currentUser._id}
                token={store.currentUser.token}
                email={store.currentUser.email}
                updateUserDetails={mockedUpdateUserDetails}
            />
        );
        expect(mockedUpdateUserDetails).not.toHaveBeenCalled();
        container.first().shallow().find('form').simulate('submit', {preventDefault: () => {} });
        expect(mockedUpdateUserDetails).toHaveBeenCalledTimes(1);
        expect(mockedUpdateUserDetails).toHaveBeenCalledWith(
            {
                username: store.users.user_A.username,
                email: store.currentUser.email,
                bio: store.users.user_A.bio,
            }, 
            store.currentUser._id, 
            store.currentUser.token
        );

    });
});