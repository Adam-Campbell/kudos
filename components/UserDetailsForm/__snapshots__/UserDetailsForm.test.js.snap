// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetailsFormContainer/> renders correctly 1`] = `
<UserDetailsForm
  bio="The bio for User A."
  email="example@example.com"
  handleBioUpdate={[Function]}
  handleEmailUpdate={[Function]}
  handleSubmit={[Function]}
  handleUsernameUpdate={[Function]}
  username="Joe Bloggs"
/>
`;

exports[`<UserDetailsFormContainer/> renders correctly 2`] = `
<Wrapper
  tight={true}
>
  <FormContainer>
    <form
      onSubmit={[Function]}
    >
      <Fieldset>
        <Legend>
          Update your account details
        </Legend>
        <Label
          htmlFor="username"
        >
          Username:
        </Label>
        <Input
          id="username"
          name="username"
          onChange={[Function]}
          type="text"
          value="Joe Bloggs"
        />
        <Label
          htmlFor="email"
        >
          Email:
        </Label>
        <Input
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
          value="example@example.com"
        />
        <Label
          htmlFor="bio"
        >
          Bio
        </Label>
        <Textarea
          id="bio"
          onChange={[Function]}
          value="The bio for User A."
        />
      </Fieldset>
      <Button
        type="submit"
      >
        Save
      </Button>
    </form>
  </FormContainer>
</Wrapper>
`;
